cpp/include/cudf/fixed_point/fixed_point.hpp

```diff
@@ -72,6 +72,371 @@ constexpr inline auto is_supported_representation_type()
 // Helper functions for `fixed_point` type
 namespace detail {
 
+/**
+ * @brief Recursively calculate a signed large power of 10 (>= 10^18) that can only be stored in an
+ * 128bit integer
+ *
+ * @note Intended to be run at compile time.
+ *
+ * @tparam Exp10 The power of 10 to calculate
+ * @return Returns 10^Exp10
+ */
+template <int Exp10>
+constexpr __int128_t large_power_of_10()
+{
+  // Stop at 10^18 to speed up compilation; literals can be used for smaller powers of 10.
+  static_assert(Exp10 >= 18);
+  if constexpr (Exp10 == 18)
+    return __int128_t(1000000000000000000LL);
```

Comment:
Use separators for readability.
```suggestion
    return __int128_t(1'000'000'000'000'000'000LL);
```

Reply:
The jitify compile cannot handle separators when combined with the LL at the end.

Reply:
Have you opened a bug report for that?

Reply:
No, should I open an issue here on the github repo, or a bug through the nvidia bugs system?

Reply:
https://github.com/rapidsai/cudf/issues/15602

Reply:
Ah ok, [done](https://github.com/NVIDIA/jitify/issues/135).